### ISSUE TITLE
Context of non-debugger source panes should be the class

### ISSUE THREAD

**Issue by ericwinger** [ericwinger/Jade#517]
Currently, executing the commented text ``self methodDoSomething`` gets the walkback below. Need to change the context of project browser/method list source pane executions to be the class. 
```
methodDoSomething

	" self methodDoSomething "
	^1
```
```
a MessageNotUnderstood occurred (error 2010), a UndefinedObject does not understand  #'methodDoSomething'

[1] MessageNotUnderstood >> defaultAction @2 line 3
[2] MessageNotUnderstood (AbstractException) >> _signalWith: @5 line 25
[3] MessageNotUnderstood (AbstractException) >> signal @2 line 47
[4] UndefinedObject (Object) >> doesNotUnderstand: @9 line 10
[5] UndefinedObject (Object) >> _doesNotUnderstand:args:envId:reason: @7 line 12
[6] Executed Code  @2 line 1
[7] String >> evaluateInContext:symbolList: @3 line 9
[8] [] in  ExecBlock0 (RowanAnsweringService) >> exec:context: @5 line 3
[9] ExecBlock0 (ExecBlock) >> on:do: @3 line 42
[10] RowanAnsweringService >> exec:context: @2 line 4
[11] RowanAnsweringService (RowanService) >> servicePerform:withArguments: @5 line 5
[12] [] in  ExecBlock1 (JadeServer) >> updateFromSton: @9 line 12
[13] OrderedCollection (Collection) >> do: @5 line 10
[14] [] in  ExecBlock0 (JadeServer) >> updateFromSton: @2 line 8
[15] ExecBlock0 (ExecBlock) >> on:do: @3 line 42
[16] JadeServer64bit32 (JadeServer) >> updateFromSton: @8 line 13
[17] GsNMethod class >> _gsReturnToC @1 line 1
===============================================
[1] MessageNotUnderstood >> defaultAction @2 line 3
	receiver oop(74719489) == a MessageNotUnderstood occurred (error 2010), a UndefinedObject does not understand  #'methodDoSomething'

instance variables:
	-receiver oop(20) == nil
	-gsResumable oop(268) == true
	-gsNumber oop(16082) == 2010
	-currGsHandler oop(20) == nil
	-gsStack oop(20) == nil
	-gsReason oop(20) == nil
	-gsTrappable oop(268) == true
	-tag oop(20) == nil
	-messageText oop(74720769) == 'a MessageNotUnderstood occurred (error 2010), a UndefinedObject does not understand  #''methodDoSomething'''
	-gsArgs oop(74719745) == anArray( )
	-selector oop(43396353) == #'methodDoSomething'
	-envId oop(2) == 0
	-gsDetails oop(20) == nil

temporary variables and parameters:
	result oop(20) == nil
----------------------------------------------------
[2] MessageNotUnderstood (AbstractException) >> _signalWith: @5 line 25
	receiver oop(74719489) == a MessageNotUnderstood occurred (error 2010), a UndefinedObject does not understand  #'methodDoSomething'

instance variables:
	-receiver oop(20) == nil
	-gsResumable oop(268) == true
	-gsNumber oop(16082) == 2010
	-currGsHandler oop(20) == nil
	-gsStack oop(20) == nil
	-gsReason oop(20) == nil
	-gsDetails oop(20) == nil
	-gsTrappable oop(268) == true
	-messageText oop(74720769) == 'a MessageNotUnderstood occurred (error 2010), a UndefinedObject does not understand  #''methodDoSomething'''
	-gsArgs oop(74719745) == anArray( )
	-selector oop(43396353) == #'methodDoSomething'
	-envId oop(2) == 0
	-tag oop(20) == nil

temporary variables and parameters:
	inCextensionArg oop(20) == nil
	res oop(20) == nil
----------------------------------------------------
[3] MessageNotUnderstood (AbstractException) >> signal @2 line 47
	receiver oop(74719489) == a MessageNotUnderstood occurred (error 2010), a UndefinedObject does not understand  #'methodDoSomething'

instance variables:
	-envId oop(2) == 0
	-gsDetails oop(20) == nil
	-gsNumber oop(16082) == 2010
	-currGsHandler oop(20) == nil
	-gsStack oop(20) == nil
	-gsReason oop(20) == nil
	-gsTrappable oop(268) == true
	-tag oop(20) == nil
	-messageText oop(74720769) == 'a MessageNotUnderstood occurred (error 2010), a UndefinedObject does not understand  #''methodDoSomething'''
	-gsArgs oop(74719745) == anArray( )
	-selector oop(43396353) == #'methodDoSomething'
	-receiver oop(20) == nil
	-gsResumable oop(268) == true

temporary variables and parameters:
----------------------------------------------------
[4] UndefinedObject (Object) >> doesNotUnderstand: @9 line 10
	receiver oop(20) == nil

instance variables:

temporary variables and parameters:
	aMessageDescriptor oop(75384321) == anArray( #'methodDoSomething', anArray( ))
	args oop(74719745) == anArray( )
	ex oop(74719489) == a MessageNotUnderstood occurred (error 2010), a UndefinedObject does not understand  #'methodDoSomething'
	sel oop(43396353) == #'methodDoSomething'
----------------------------------------------------
[5] UndefinedObject (Object) >> _doesNotUnderstand:args:envId:reason: @7 line 12
	receiver oop(20) == nil

instance variables:

temporary variables and parameters:
	aSymbol oop(43396353) == #'methodDoSomething'
	anArray oop(74719745) == anArray( )
	aSmallInt oop(2) == 0
	dnuKind oop(2) == 0
----------------------------------------------------
[6] Executed Code  @2 line 1
	receiver oop(20) == nil

instance variables:

temporary variables and parameters:
----------------------------------------------------
[7] String >> evaluateInContext:symbolList: @3 line 9
	receiver oop(75326977) == 'self methodDoSomething'

instance variables:

temporary variables and parameters:
	anObject oop(20) == nil
	aSymbolList oop(75326721) == aSymbolList( aSymbolDictionary( #'UserGlobals'->aSymbolDictionary, #'RwUserPlatform_Preferences'->aDictionary( #'automaticClassInitializationBlackList'->anOrderedCollection( )), #'def'->$a, #'JadeServ...
----------------------------------------------------
[8] [] in  ExecBlock0 (RowanAnsweringService) >> exec:context: @5 line 3
	receiver oop(74972417) == anExecBlock0

instance variables:

temporary variables and parameters:
	aString oop(75326977) == 'self methodDoSomething'
	oop oop(162) == 20
----------------------------------------------------
[9] ExecBlock0 (ExecBlock) >> on:do: @3 line 42
	receiver oop(74972417) == anExecBlock0

instance variables:

temporary variables and parameters:
	exceptionSelector oop(132097) == CompileError
	fastPath oop(268) == true
	handlerBlock oop(38164993) == anExecBlock1
----------------------------------------------------
[10] RowanAnsweringService >> exec:context: @2 line 4
	receiver oop(75274753) == aRowanAnsweringService

instance variables:
	-answer oop(20) == nil
	-definition oop(20) == nil
	-command oop(35900673) == #'exec:context:'
	-commandArgs oop(75275009) == anArray( 'self methodDoSomething', 20)
	-updates oop(20) == nil
	-organizer oop(75275265) == aClassOrganizer
	-updateType oop(20) == nil

temporary variables and parameters:
	aString oop(75326977) == 'self methodDoSomething'
	oop oop(162) == 20
----------------------------------------------------
[11] RowanAnsweringService (RowanService) >> servicePerform:withArguments: @5 line 5
	receiver oop(75274753) == aRowanAnsweringService

instance variables:
	-answer oop(20) == nil
	-definition oop(20) == nil
	-command oop(35900673) == #'exec:context:'
	-commandArgs oop(75275009) == anArray( 'self methodDoSomething', 20)
	-updates oop(20) == nil
	-organizer oop(75275265) == aClassOrganizer
	-updateType oop(20) == nil

temporary variables and parameters:
	collection oop(75275009) == anArray( 'self methodDoSomething', 20)
	symbol oop(35900673) == #'exec:context:'
----------------------------------------------------
[12] [] in  ExecBlock1 (JadeServer) >> updateFromSton: @9 line 12
	receiver oop(75274497) == anExecBlock1

instance variables:

temporary variables and parameters:
	organizer oop(75275265) == aClassOrganizer
	service oop(75274753) == aRowanAnsweringService
----------------------------------------------------
[13] OrderedCollection (Collection) >> do: @5 line 10
	receiver oop(75208449) == anOrderedCollection( aRowanLoggingService, aRowanAnsweringService)

instance variables:

temporary variables and parameters:
	aBlock oop(75274497) == anExecBlock1
	i oop(18) == 2
----------------------------------------------------
[14] [] in  ExecBlock0 (JadeServer) >> updateFromSton: @2 line 8
	receiver oop(75195649) == anExecBlock0

instance variables:

temporary variables and parameters:
	services oop(75208449) == anOrderedCollection( aRowanLoggingService, aRowanAnsweringService)
----------------------------------------------------
[15] ExecBlock0 (ExecBlock) >> on:do: @3 line 42
	receiver oop(75195649) == anExecBlock0

instance variables:

temporary variables and parameters:
	exceptionSelector oop(36371969) == GsInteractionRequest
	fastPath oop(268) == true
	handlerBlock oop(75425281) == anExecBlock
----------------------------------------------------
[16] JadeServer64bit32 (JadeServer) >> updateFromSton: @8 line 13
	receiver oop(73731073) == aJadeServer64bit32

instance variables:
	-environment oop(2) == 0
	-classList oop(20) == nil
	-readStream oop(20) == nil
	-writeStream oop(75191809) == aWriteStream
	-selectedClass oop(20) == nil
	-methodFilterType oop(20) == nil
	-methodFilters oop(20) == nil
	-classOrganizers oop(73871873) == anArray( nil, nil, nil, nil)
	-methodCommandResult oop(20) == nil
	-selections oop(20) == nil

temporary variables and parameters:
	organizer oop(75275265) == aClassOrganizer
	resultString oop(20) == nil
	stonString oop(74717953) == 'OrderedCollection[RowanLoggingService{#command:#logReceivedServices,#commandArgs:[],#id:0,#groupId:67249,#services:OrderedCollection[RowanAnsweringService{#command:#''exec:context:'',#commandArgs:[''...
	self oop(73731073) == aJadeServer64bit32
	services oop(75208449) == anOrderedCollection( aRowanLoggingService, aRowanAnsweringService)
----------------------------------------------------
[17] GsNMethod class >> _gsReturnToC @1 line 1
	receiver oop(144897) == GsNMethod

instance variables:

temporary variables and parameters:
```

**Comment by ericwinger** [comment]
Note that this is in a project browser. Same fix should apply to the method browser as well.

**Comment by ericwinger** [comment]
Fixed in 3.0.78.

**Comment by LisaAlmarode** [comment]
Looks nice, but it is true that the method context is not available in the debugger though not quite so clear why.  I like the method context in the debugger in GBS.
